keanu-project/src/main/java/io/improbable/keanu/util/csv/WriteCsv.java

```diff
@@ -0,0 +1,124 @@
+package io.improbable.keanu.util.csv;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.vertices.Vertex;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class provides static helper functions for writing csv data
+ */
+public class WriteCsv {
+
+    private static String DEFAULT_EMPTY_VALUE = "-";
+
+    private WriteCsv() {
+    }
+
+    /**
+     * @param samples the samples to be written to CSV
+     * @param vertices the vertices whose samples will be written to CSV
+     * @param <T> the type of the vertices
+     * @return a writer for the csv file
+     */
+    public static <T extends Tensor> CsvWriter asSamples(NetworkSamples samples, List<Vertex<T>> vertices) {
```

Comment:
This generic means that this method can only be used on lists of vertices of all the same type. We might want to write mixed data types right?